coffee-module/coffee-module-redisstream/src/main/java/hu/icellmobilsoft/coffee/module/redisstream/publisher/RedisStreamPublisher.java

```diff
@@ -207,6 +211,46 @@ public List<Optional<StreamEntryID>> publishPublications(List<RedisStreamPublica
         }
     }
 
+    /**
+     * Publish (send) multiple messages through pipeline to stream calculated by input publication streamGroup name.
+     *
+     * @param publications
+     *            stream publication data list
+     * @return Created Redis Stream messages identifiers from Redis server
+     * @throws BaseException
+     *             exception on sending
+     */
+    public List<Optional<StreamEntryID>> publishPublicationsPipelined(List<RedisStreamPublication> publications) throws BaseException {
+        if (publications == null) {
+            throw new InvalidParameterException("publications is null!");
+        }
+        checkRedisManager();
+
+        try (RedisManagerConnection ignored = redisManager.initConnection()) {
+            List<Response<StreamEntryID>> responses = new ArrayList<>(publications.size());
+            Pipeline pipeline = initPipeline();
+            for (RedisStreamPublication publication : publications) {
+                Response<StreamEntryID> response;
+                if (StringUtils.isBlank(publication.getStreamGroup())) {
+                    validateGroup(streamGroup);
+                    response = publishThroughPipeline(
+                            pipeline,
+                            streamGroup,
+                            createJedisMessage(publication.getStreamMessage(), publication.getParameters()));
+                } else {
+                    validateGroup(publication.getStreamGroup());
+                    response = publishThroughPipeline(
+                            pipeline,
+                            publication.getStreamGroup(),
+                            createJedisMessage(publication.getStreamMessage(), publication.getParameters()));
+                }
+                responses.add(response);
+            }
+            pipeline.sync();
```

Comment:
Jó lenne minden 100 vagy 1000 elem után sync-elni, hogy ha sok elemű a lista, ne egyben legyen ráküldve (addig is dolgozzon a redis). A BatchService-nél is van [batchSize kezelés](https://github.com/i-Cell-Mobilsoft-Open-Source/frappee/blob/main/frappee-hibernate/frappee-hibernate-batch/src/main/java/hu/icellmobilsoft/frappee/hibernate/batch/HibernateBatchService.java#L300).

Reply:
Nekem ahogy teszteltem, 10000 message esetén gyorsabb volt, mikor minden 1000 után synceltem, mint mikor minden 100. után, úgyhogy 1000 lett. De felülírható úgyis

Reply:
gyorsabbnak gyorsabb lehet, Martin valószínűleg azért is kérdezi mert ezeket a műveleteket a redis szerver sorba rendezi, ami meg a memoria használatát mövelheti meg redis szervernek

Reply:
Az 1000 egy teljesen jó szám (pláne amilyen kis üzeneteket küldünk még nagyobb is lehetne). Csak annyi a lényeg, hogy legyen valami korlát, és ha 100k-t küld be valaki, akkor az ne egyszerre terhelje a redis-t (cpu, memória, stb...)